protected/humhub/modules/content/components/ActiveQueryContent.php

```diff
@@ -58,18 +58,25 @@ public function readable($user = null)
         $globalCondition = '';
 
         if ($user !== null) {
-            $this->leftJoin('space_membership', 'contentcontainer.pk=space_membership.space_id AND contentcontainer.class=:spaceClass AND space_membership.user_id=:userId', [':userId' => $user->id, ':spaceClass' => Space::class]);
+            if ($user->canViewAllContent()) {
+                // Don't restrict if user can view all content:
+                $conditionSpaceMembershipRestriction = '';
+                $conditionUserPrivateRestriction = '';
+            } else {
+                // User must be a space's member OR Space and Content are public
+                $this->leftJoin('space_membership', 'contentcontainer.pk=space_membership.space_id AND contentcontainer.class=:spaceClass AND space_membership.user_id=:userId', [':userId' => $user->id, ':spaceClass' => Space::class]);
```

Comment:
Note that this is a breaking change since the membership join is now not set anymore by default. It may have side effects since other queries may depend on this join.

Reply:
ok, I moved the joins back, probably you are right, some queries may depend on this, I thought the join is used only inside the method, it is why I put it under the condition.

Reply:
Yeah, its not obvoius but the `Content::readable()` is the basis of many other filters etc.